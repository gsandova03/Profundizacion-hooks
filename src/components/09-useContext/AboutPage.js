import React from 'react'

export const AboutPage = () => {
  return (
    <div>
      <h4>About Page</h4>
      <hr/>
    </div>
  )
}
