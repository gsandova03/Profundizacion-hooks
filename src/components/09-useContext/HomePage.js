import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomePage = () => {

  const { user } = useContext( UserContext )
  console.log( user )
  return (
    <div>
      <h4>Home Page</h4>
      <hr/>
      <pre>
        { JSON.stringify( user, null, 3 ) }
      </pre>

    </div>
  )
}
