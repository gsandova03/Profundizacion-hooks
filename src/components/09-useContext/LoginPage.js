import React, { useContext } from 'react'
import { useForm } from '../../hooks/useForm'
import { UserContext } from './UserContext'

export const LoginPage = () => {

  const { setUser } = useContext( UserContext )
  return (
    <div>
      <h4>Login Page</h4>
      <hr/>
      <button 
        type="submit" 
        className="btn btn-primary"
        onClick={ ()=> setUser({
          id: 1242,
          name: "gabriel"
        })}
      > 
        Login 
      </button>
    </div>
  )
}
