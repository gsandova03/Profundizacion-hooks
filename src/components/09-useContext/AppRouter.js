import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { Navbar } from './components/Navbar';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/about" component={ AboutPage }/>
          <Route exact path="/login" component={ LoginPage }/>
          <Route exact path="/" component={ HomePage }/>
          <Redirect to="/"/>
        </Switch>
      </div>
    </Router>
  )
}
