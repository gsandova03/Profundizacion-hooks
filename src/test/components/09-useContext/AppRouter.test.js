import React from 'react'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext'
import { AppRouter } from '../../../components/09-useContext/AppRouter'


describe('Pruebas en el componente <AppRouter/>', () => {
  
  const user = {
    id:11,
    name: "gabriel",
  }

  const wrapper = mount(
    <UserContext.Provider value={ user }>
      <AppRouter/>
    </UserContext.Provider>
  )

  test('debe de mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  })
  
})
