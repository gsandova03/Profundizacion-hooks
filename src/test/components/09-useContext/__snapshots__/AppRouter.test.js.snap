// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AppRouter/> debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark mb-5"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
         useContext 
      </a>
      <div
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h4>
      Home Page
    </h4>
    <hr />
    <pre />
  </div>
</div>
`;
